eng/common/scripts/Package-Properties.ps1

```diff
@@ -79,6 +82,61 @@ class PackageProps
         $this.Initialize($name, $version, $directoryPath, $serviceDirectory)
         $this.Group = $group
     }
+
+    hidden [object]GetValueSafely($hashtable, [string[]]$keys) {
+        $current = $hashtable
+        foreach ($key in $keys) {
+            if ($current.ContainsKey($key) -or $current[$key]) {
+                $current = $current[$key]
+            }
+            else {
+                return $null
+            }
+        }
+
+        return $current
+    }
+
+    hidden [HashTable]ParseYmlForArtifact([string]$ymlPath) {
+        if (Test-Path -Path $ymlPath) {
+            try {
+                $content = Get-Content -Raw -Path $ymlPath | CompatibleConvertFrom-Yaml
+                if ($content) {
+                    $artifacts = $this.GetValueSafely($content, @("extends", "parameters", "Artifacts"))
+
+                    $artifactForCurrentPackage = $artifacts | Where-Object { $_["name"] -eq $this.ArtifactName -or $_["name"] -eq $this.Name }
+
+                    if ($artifactForCurrentPackage) {
+                        return [HashTable]$artifactForCurrentPackage
+                    }
+                }
+            }
+            catch {
+              Write-Host "Exception while parsing yml file $($ymlPath): $_"
+            }
+        }
+
+        return $null
+    }
+
+    [void]InitializeCIArtifacts(){
+        $RepoRoot = Resolve-Path (Join-Path $PSScriptRoot ".." ".." "..")
+        # $ciFilePath = Join-Path -Path $RepoRoot -ChildPath (Join-Path "sdk" $this.ServiceDirectory "ci.yml")
```

Comment:
I assume these commented out lines can just be removed.

Reply:
damnit I thought I cleaned everything. **shakes fist**